sdk/cosmosdb/cosmos/src/client/ChangeFeed/changeFeedUtils.ts

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { ChangeFeedIteratorOptions } from "./ChangeFeedIteratorOptions";
+import { ErrorResponse } from "../../request";
+import { PartitionKeyRange } from "../Container";
+import { InternalChangeFeedIteratorOptions } from "./InternalChangeFeedOptions";
+import { isPrimitivePartitionKeyValue } from "../../utils/typeChecks";
+import { ChangeFeedStartFrom } from "./ChangeFeedStartFrom";
+import { ChangeFeedStartFromBeginning } from "./ChangeFeedStartFromBeginning";
+import { ChangeFeedStartFromNow } from "./ChangeFeedStartFromNow";
+import { Constants } from "../../common";
+import { ChangeFeedStartFromTime } from "./ChangeFeedStartFromTime";
+import { QueryRange } from "../../routing";
+import { FeedRangeInternal } from "./FeedRange";
+
+/**
+ * @hidden
+ * Validates the change feed options passed by the user
+ */
+export function validateChangeFeedIteratorOptions(options: ChangeFeedIteratorOptions): void {
+  if (!isChangeFeedIteratorOptions(options)) {
+    throw new ErrorResponse("Invalid Changefeed Iterator Options.");
+  }
+  if (options?.maxItemCount && typeof options?.maxItemCount !== "number") {
+    throw new ErrorResponse("maxItemCount must be number");
+  }
+  if (options?.maxItemCount !== undefined && options?.maxItemCount < 1) {
+    throw new ErrorResponse("maxItemCount must be a positive number");
+  }
+}
+
+function isChangeFeedIteratorOptions(options: unknown): options is ChangeFeedIteratorOptions {
+  if (Object.keys(options).length === 0) {
+    return true;
+  }
+  return options && !(isPrimitivePartitionKeyValue(options) || Array.isArray(options));
+}
+
+/**
+ * @hidden
+ * Checks if pkRange entirely covers the given overLapping range or there is only partial overlap.
+ *
+ * If no complete overlap, exact range which overlaps is retured which is used to set minEpk and maxEpk headers while quering change feed.
+ */
+export async function checkEpkHeaders(
+  epkRange: QueryRange,
+  overLappingRange: PartitionKeyRange
+): Promise<[string, string]> {
+  if (
+    overLappingRange.minInclusive >= epkRange.min &&
+    overLappingRange.maxExclusive <= epkRange.max
+  ) {
+    return [undefined, undefined];
+  } else if (
+    overLappingRange.minInclusive <= epkRange.min &&
+    overLappingRange.maxExclusive >= epkRange.max
+  ) {
+    return [epkRange.min, epkRange.max];
+  }
+  // Right Side of overlapping range is covered
+  else if (
+    overLappingRange.minInclusive <= epkRange.min &&
+    overLappingRange.maxExclusive <= epkRange.max &&
+    overLappingRange.maxExclusive >= epkRange.min
+  ) {
+    return [epkRange.min, overLappingRange.maxExclusive];
+  }
+  // Left Side of overlapping range is covered
+  else {
+    return [overLappingRange.minInclusive, epkRange.max];
+  }
+}
+
+/**
+ * @hidden
+ * Checks if the object is a valid EpkRange
+ */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
```

Comment:
I think if we use `unknown` instead of `any`, we wouldn't need to suppress warning.

Reply:
Done